packages/compartment-mapper/src/import-archive.js

```diff
@@ -264,6 +285,7 @@ export const parseArchive = async (
       archiveLocation,
       computeSha512,
       computeSourceLocation,
+      exitModuleImportHook,
```

Comment:
I don't enjoy passing the hook to the archive making part of the archive flow. 
It should not be necessary before runtime. 

Solutions:

1. A boolean configuration value saying to "expectExitModuleImportHook" would be enough to omit the archive-time errors where exit import hook would kick in.   
pros: no unnecessary importing and hook mechanics before runtime  
cons: no early errors if exitModuleImportHook doesn't provide a module, another config item

2. exitModuleImportHook accepts options, including `archiveOnly` and if that's truthy, it can avoid importing actual modules while still returning undefined where it originally would (eg. if checking against a list of known builtins it wishes to support)  
pros: no additional option, no need to import actual modules during archiving  
cons: more complexity in exitModuleImportHook

Reply:
I would rename this parameter to `isExitModuleImportAllowed`, since the logic for it may be distinct from the logic of `exitModuleImportHook`, and it only needs to return truthy or falsy (default: `() => false`).  Using `myExitModuleImportHook` for the parameter is a degenerate case that works, but isn't ideal (it does too much work, as you point out above).